src/parser/core/modules/Positionals.tsx

```diff
@@ -0,0 +1,236 @@
+import {t} from '@lingui/macro'
+import {Trans} from '@lingui/react'
+import {ActionLink, DataLink} from 'components/ui/DbLink'
+import {Action} from 'data/ACTIONS'
+import {Cause, Event, Events} from 'event'
+import {filter, oneOf} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import Checklist, {Requirement, Rule} from 'parser/core/modules/Checklist'
+import DISPLAY_ORDER from 'parser/core/modules/DISPLAY_ORDER'
+import {Timeline} from 'parser/core/modules/Timeline'
+import React from 'react'
+import {Button, Table} from 'semantic-ui-react'
+import {Analyser} from '../Analyser'
+import {Data} from './Data'
+
+export interface PositionalResult {
+	positional: Positional,
+	hits: Array<Events['damage']>,
+	misses: Array<Events['damage']>
+}
+
+// The commented out enums are unused because only plain combo actions
+// are required for the positional checking to work for now.
+// "NONE" or "BASE" was considered as a modifier, but it could result in
+// invalid states like [NONE, COMBO] that need to be checked.
+export enum PotencyModifier {
+	COMBO,
+	// POSITIONAL,
+	// DRG_LANCE_MASTERY,
+	// RPR_ENHANCED_REAVE,
+}
+
+// Potency is modeled this way because any single potency number
+// can have a combination of states that apply to it, see all
+// of the commented out PotencyModifiers. An empty modifier
+// list means it's the base potency.
+interface Potency {
+	value: number,
+	modifiers: PotencyModifier[]
+}
+
+// The absence of potencies current means that it does NOT have combo actions,
+// because in order to detect positionals, checking combo non-positional
+// potencies is sufficient. For full potency modeling, potencies
+// should not be optional.
+interface Positional {
+	action: Action,
+	potencies?: Potency[]
+}
+
+const NO_BONUS_PERCENT = 0
+
+export abstract class Positionals extends Analyser {
+	@dependency protected data!: Data
+	@dependency private checklist!: Checklist
+	@dependency private timeline!: Timeline
+
+	static override handle = 'positionals'
+	static override title = t('core.positionals.title')`Positionals`
+	static override displayOrder = DISPLAY_ORDER.POSITIONALS
+
+	private positionalResults: PositionalResult[] = []
+
+	/**
+	 * Jobs MUST provide a list of their positional actions
+	 */
+	protected abstract positionals: Positional[]
+
+	override initialise() {
+		this.addEventHook(
+			filter<Event>().source(this.parser.actor.id).type('damage')
+				.cause(filter<Cause>().action(oneOf(this.positionals.map(positional => positional.action.id)))), this.onCast)
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private onCast(event: Events['damage']) {
+		if (event.cause.type !== 'action') {
+			return
+		}
+		const action = this.data.getAction(event.cause.action)
+		if (!action) {
```

Comment:
```suggestion
		if (action == null) {
```
Style guide: We prefer explicit == null checks for null/undefined guards, rather than relying on falsey behavior